File: backend/src/german_app/i18n/languages.py
from typing import Literal, TypedDict


class LanguageInfo(TypedDict):
    label: str
    speech_locale: str


LanguageCode = Literal["de", "en", "fr", "ja", "pt", "es"]


SUPPORTED_LANGUAGES: dict[str, LanguageInfo] = {
    "de": {"label": "Deutsch", "speech_locale": "de-DE"},
    "en": {"label": "English", "speech_locale": "en-US"},
    "fr": {"label": "Français", "speech_locale": "fr-FR"},
    "ja": {"label": "日本語", "speech_locale": "ja-JP"},
    "pt": {"label": "Português", "speech_locale": "pt-PT"},
    "es": {"label": "Español", "speech_locale": "es-ES"},
}


def validate_language(code: str) -> str:
    if code not in SUPPORTED_LANGUAGES:
        raise ValueError(
            f"Unsupported language code '{code}'. Supported: {sorted(SUPPORTED_LANGUAGES)}"
        )
    return code


def language_label(code: str) -> str:
    return SUPPORTED_LANGUAGES.get(code, {"label": code, "speech_locale": code}).get(
        "label", code
    )


def speech_locale(code: str) -> str:
    return SUPPORTED_LANGUAGES.get(code, {"label": code, "speech_locale": code}).get(
        "speech_locale", code
    )

File: frontend/src/lib/languages.ts
export type LanguageCode = "de" | "en" | "fr" | "ja" | "pt" | "es";

export interface LanguageInfo {
  label: string;
  speechLocale: string;
}

export const SUPPORTED_LANGUAGES: Record<LanguageCode, LanguageInfo> = {
  de: { label: "Deutsch", speechLocale: "de-DE" },
  en: { label: "English", speechLocale: "en-US" },
  fr: { label: "Français", speechLocale: "fr-FR" },
  ja: { label: "日本語", speechLocale: "ja-JP" },
  pt: { label: "Português", speechLocale: "pt-PT" },
  es: { label: "Español", speechLocale: "es-ES" },
};

export const LANGUAGE_CODES: LanguageCode[] = ["de", "en", "fr", "ja", "pt", "es"];

export function languageLabel(code: string): string {
  return SUPPORTED_LANGUAGES[code as LanguageCode]?.label ?? code;
}

export function speechLocale(code: string): string {
  return SUPPORTED_LANGUAGES[code as LanguageCode]?.speechLocale ?? code;
}
